### PR TITLE
16/edit cap

### PR DESCRIPTION
**Pull request by Natino64**
## Description
Changes made for this branch

## Checklist
- [x] All manual tests passing
- [x] All unit tests passing 
- [x] All UI tests passing
- [x] Code reviewed by peer:\
&nbsp; - *No redundant whitespace*\
&nbsp; - *Indentation correct*\
&nbsp; - *Naming conventions followed*\
&nbsp; - *Comments added (where needed)*\
&nbsp; - *No commented out redundant code*\
&nbsp; - *Branch name correct*
- [ ] Code reviewed by TA

### CODE REVIEW

**Comment by jurassic-marc** [comment]
Capabilities fields need to make clickable

Static modals

File: Angular/src/app/admin-table/admin-table.component.ts [hunk 1]
```diff
@@ -93,6 +100,11 @@ export class AdminTableComponent implements OnInit {
     this.dataService.getDepartmentDetails(this.departmentId).then(response => {
       this.departmentName = response;
     });
+
+    this.dataService.getAllDepartments().then(response => {
+      this.departments = response;
+      console.log(this.departments);
```
Comment (matthewdoreilly) [inline Angular/src/app/admin-table/admin-table.component.ts]: Remove this log

File: Angular/src/app/edit-capability-modal/edit-capability-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  writeToDatabase() {
+         this.dataService
+        .editCapability(this.editedCapability)
+        .then(result => this.capabilityEdited.emit(result))
+        .catch(error => {
+          this.capabilityEdited.emit(error);
+        });
+      this.closeModal();
+  }
```
Comment (matthewdoreilly) [inline Angular/src/app/edit-capability-modal/edit-capability-modal.component.ts]: Fix the indentation here, like:

```
writeToDatabase() {
     this.dataService.editCapability(this.editedCapability)
          .then(
               result => this.capabilityEdited.emit(result)
          )
          .catch(error => {
               this.capabilityEdited.emit(error);
          });
     this.closeModal();
}
```

File: Angular/src/app/services/modal.service.ts [hunk 3]
```diff
@@ -88,4 +89,18 @@ export class ModalService {
       });
     })
   }
+
+    async openEditCapabilityModal(capabilityId, departments) {
+    return new Promise((resolve, reject) => {
+      this.dataService.getEditCapability(capabilityId).then(response => {
+        const modalRef = this.modalService.open(EditCapabilityModalComponent, { backdrop: "static" });
+        modalRef.componentInstance.capabilityToEdit = response[0];
+        console.log(response[0])
```
Comment (matthewdoreilly) [inline Angular/src/app/services/modal.service.ts]: Remove this log

File: Express-API/department.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 const getDepartmentsQuery = "SELECT name FROM department WHERE department_id = ?"
+const allDepartmentsQuery = "SELECT * FROM department"
```
Comment (matthewdoreilly) [inline Express-API/department.js]: Never `SELECT *`. Only select the columns from the `department` table which you need

**Pull request merged**